src/components/DataTable/DataTable.tsx

```diff
@@ -328,7 +328,7 @@ export class DataTable extends React.PureComponent<CombinedProps, State> {
     } = this;
     if (collapsed && table && scrollContainer && dataTable) {
       const headerCells = table.querySelectorAll(
-        '[class*=header]',
+        '[data-polaris-headercell]',
```

Comment:
```suggestion
        '[data-polaris-header-cell]',
```

Reply:
Could you add the attribute to this file https://github.com/Shopify/polaris-react/blob/master/src/components/shared.ts 🙏